File: backend/src/main/java/com/smartcampus/backend/dto/search/SearchResultDTO.java
package com.smartcampus.backend.dto.search;

/**
 * Représente un résultat de recherche de lieu (géocodage).
 * Miroir exact de SearchResultDto.kt côté Android :
 * - name      : titre court (ex: "Kedge Business School")
 * - subtitle  : complément d'adresse (ex: "Talence, Gironde")
 * - latitude / longitude : coordonnées pour positionner le marqueur
 */
public class SearchResultDTO {

    private String name;
    private String subtitle;
    private double latitude;
    private double longitude;

    public SearchResultDTO() {
    }

    public SearchResultDTO(String name, String subtitle, double latitude, double longitude) {
        this.name = name;
        this.subtitle = subtitle;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSubtitle() {
        return subtitle;
    }

    public void setSubtitle(String subtitle) {
        this.subtitle = subtitle;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/SearchResultDto.kt
package com.ObservatoireCampus.mobile.model.search

/**
 * Représente un résultat de recherche de lieu (autocomplétion).
 * - name : titre court affiché en gras (ex: "Kedge Business School")
 * - subtitle : complément d'adresse affiché en gris, plus petit (ex: "Talence, Gironde")
 * Utilisé pour positionner la caméra et le marqueur sur la carte.
 */
data class SearchResultDto(
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val subtitle: String = ""
)